sklearn/model_selection/_search.py

```diff
@@ -560,44 +583,91 @@ def _fit(self, X, y, labels, parameter_iterable):
+            search_results["%s_split_%s" % (metric, split_i)] = (
+                np.empty(res_shape, dtype=np.float64))
+        search_results["%s_mean" % metric] = np.empty(res_shape,
+                                                      dtype=np.float64)
+        search_results["%s_rank" % metric] = np.empty(res_shape, dtype=int)
+
+        for fit_i, grid_start in enumerate(range(0, n_fits, n_splits)):
+            n_test_samples_total = 0
+            mean_score = 0
+
+            split_i = -1
+            for score_i, n_test_samples_i, _, parameters in \
                     out[grid_start:grid_start + n_splits]:
-                all_scores.append(this_score)
+                split_i += 1
+                # Record the score/n_test_samples for the i-th split
+                # of the current parameter setting candidate.
+                all_scores[split_i] = score_i
+
                 if self.iid:
-                    this_score *= this_n_test_samples
-                    n_test_samples += this_n_test_samples
-                score += this_score
+                    score_i *= n_test_samples_i
+                    n_test_samples_total += n_test_samples_i
+
+                mean_score += score_i
+                search_results["%s_split_%s" %
+                               (metric, split_i)][fit_i] = score_i
+
             if self.iid:
-                score /= float(n_test_samples)
+                mean_score = all_scores.sum() / float(n_test_samples_total)
             else:
-                score /= float(n_splits)
-            scores.append((score, parameters))
-            # TODO: shall we also store the test_fold_sizes?
-            grid_scores.append(_CVScoreTuple(
-                parameters,
-                score,
-                np.array(all_scores)))
-        # Store the computed scores
-        self.grid_scores_ = grid_scores
+                mean_score = all_scores.mean()
+
+            # Store the mean score and the parameters for this fit
+            search_results["%s_mean" % metric][fit_i] = mean_score
+            for param in parameters:
+                # This entry alone gets unmasked when assigned
+                search_results["param_%s" % param][fit_i] = parameters[param]
 
         # Find the best parameters by comparing on the mean validation score:
         # note that `sorted` is deterministic in the way it breaks ties
-        best = sorted(grid_scores, key=lambda x: x.mean_validation_score,
-                      reverse=True)[0]
-        self.best_params_ = best.parameters
-        self.best_score_ = best.mean_validation_score
+        # We reverse the order to get a descending sort order
+        sorted_indices = np.argsort(
+            search_results["%s_mean" % metric])[::-1]
+
+        search_results["%s_rank" % metric][sorted_indices] = (
+            np.arange(1, self.n_parameters_ + 2))
+
+        self.search_results_ = search_results
+
+        best = sorted_indices[0]
```

Comment:
I proposed an age ago that we store `best_index_` and define `property`s that calculate `best_params_` and `best_score_` from this.